fili-core/src/test/groovy/com/yahoo/bard/webservice/web/DataApiRequestIntervalsSpec.groovy

```diff
@@ -134,6 +135,60 @@ class DataApiRequestIntervalsSpec extends Specification {
 
     }
 
+    @Unroll
+    def "P#numPastDays D/current with 'all' granularity generates an interval of past #numPastDays days" () {
+        when: "parse string interval from request"
+        Set intervals = new TestingDataApiRequestImpl().generateIntervals(
+                String.format("P%sD/${TimeMacros.CURRENT.getName()}", numPastDays),
+                apiRequest.generateGranularity("all", granularityParser), dateTimeFormatter
+        )
+        Interval interval = intervals.first()
+        DateTime parsedEnd = DateTime.now()
+        DateTime parsedStart = parsedEnd.minusDays(numPastDays)
+
+        then: "an interval of past days is generated"
+        intervals.size() == 1
+        interval.start.year == parsedStart.year
+        interval.start.monthOfYear == parsedStart.monthOfYear
+        interval.start.dayOfYear == parsedStart.dayOfYear
+        interval.end.year == parsedEnd.year
+        interval.end.monthOfYear == parsedEnd.monthOfYear
+        interval.end.dayOfYear == parsedEnd.dayOfYear
+
+        where:
+        numPastDays | _
+        1           | _
+        2           | _
+        3           | _
+    }
+
+    @Unroll
+    def "current/P#numPastDays D with 'all' granularity generates an interval of #numPastDays days for the future" () {
```

Comment:
I'm inclined to go with "these changes don't make sense", but if they _do_ fit the established semantics, the tests for the newly valid syntax _should_ be able to be slotted into the existing tests. If not, why?